src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -687,119 +703,118 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
+            // and add it to allowed script files
+            if (modPath.EndsWith(@".ps1") || modPath.EndsWith(@".psm1"))
+            {
+                newFile = workspace.GetFile(modPath);
+                newFile.IsAnalysisEnabled = false;
+                scriptFiles.Add(newFile);
+            }
+            if (moduleInfo.NestedModules.Count > 0)
+            {
+                foreach (PSModuleInfo nestedInfo in moduleInfo.NestedModules)
                 {
-                    foreach (PSModuleInfo nestedInfo in moduleInfo.NestedModules)
+                    string nestedModPath = nestedInfo.Path;
+                    if (nestedModPath.EndsWith(@".ps1") || nestedModPath.EndsWith(@".psm1"))
                     {
-                        string nestedModPath = nestedInfo.Path;
-                        if (nestedModPath.EndsWith(@".ps1") || nestedModPath.EndsWith(@".psm1"))
-                        {
-                            newFile = workspace.GetFile(nestedModPath);
-                            newFile.IsAnalysisEnabled = false;
-                            scriptFiles.Add(newFile);
-                        }
+                        newFile = workspace.GetFile(nestedModPath);
+                        newFile.IsAnalysisEnabled = false;
+                        scriptFiles.Add(newFile);
                     }
                 }
-
-                return scriptFiles.ToArray();
             }
 
-            return new List<ScriptFile>().ToArray();
+            return scriptFiles.ToArray();
         }
 
         private SymbolReference FindDeclarationForBuiltinCommand(
             CommandInfo commandInfo,
             SymbolReference foundSymbol,
             Workspace workspace)
         {
-            SymbolReference foundDefinition = null;
-            if (commandInfo != null)
+            if (commandInfo == null)
             {
-                int index = 0;
-                ScriptFile[] nestedModuleFiles;
-
-                nestedModuleFiles =
-                    GetBuiltinCommandScriptFiles(
-                        commandInfo.Module,
-                        workspace);
+                return null;
+            }
 
-                while (foundDefinition == null && index < nestedModuleFiles.Length)
-                {
-                    foundDefinition =
-                        AstOperations.FindDefinitionOfSymbol(
-                            nestedModuleFiles[index].ScriptAst,
-                            foundSymbol);
+            ScriptFile[] nestedModuleFiles =
+                GetBuiltinCommandScriptFiles(
+                    commandInfo.Module,
+                    workspace);
 
-                    if (foundDefinition != null)
-                    {
-                        foundDefinition.FilePath = nestedModuleFiles[index].FilePath;
-                    }
+            SymbolReference foundDefinition = null;
+            for (int i = 0; i < nestedModuleFiles.Length; i++)
```

Comment:
did this need to be a `for` instead of a `foreach`?

Reply:
I think you're right. I was changing from a `while`, but was probably too conservative